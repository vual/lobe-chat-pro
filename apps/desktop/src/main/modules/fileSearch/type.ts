import { FileResult, SearchOptions } from '@/types/fileSearch';

/**
 * File Search Service Implementation Abstract Class
 * Defines the interface that different platform file search implementations need to implement
 */
export abstract class FileSearchImpl {
  /**
   * Perform file search
   * @param options Search options
   * @returns Promise of search result list
   */
  abstract search(options: SearchOptions): Promise<FileResult[]>;

  /**
   * Check search service status
   * @returns Promise indicating if service is available
   */
  abstract checkSearchServiceStatus(): Promise<boolean>;

  /**
   * Update search index
   * @param path Optional specified path
   * @returns Promise indicating operation success
   */
  abstract updateSearchIndex(path?: string): Promise<boolean>;
}
