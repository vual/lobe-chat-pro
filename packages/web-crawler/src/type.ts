export interface CrawlSuccessResult {
  content?: string;
  contentType: 'text' | 'json';
  description?: string;
  length?: number;
  siteName?: string;
  title?: string;
  url: string;
}

export interface CrawlErrorResult {
  content: string;
  errorMessage: string;
  errorType: string;
  url: string;
}

export interface FilterOptions {
  // 是否启用Readability
  enableReadability?: boolean;

  pureText?: boolean;
}

type CrawlImplType = 'naive' | 'jina' | 'browserless' | 'search1api';

type CrawlImplParams<T> = T & {
  filterOptions: FilterOptions;
};

export type CrawlImpl<Params = object> = (
  url: string,
  params: CrawlImplParams<Params>,
) => Promise<CrawlSuccessResult | undefined>;

export interface CrawlUrlRule {
  // 内容过滤配置（可选）
  filterOptions?: FilterOptions;
  impls?: CrawlImplType[];
  // URL匹配模式，仅支持正则表达式
  urlPattern: string;
  // URL转换模板（可选），如果提供则进行URL转换
  urlTransform?: string;
}
