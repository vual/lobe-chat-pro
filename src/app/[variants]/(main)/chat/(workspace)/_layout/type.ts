import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  conversation: ReactNode;
  portal: ReactNode;
  topic: ReactNode;
}
