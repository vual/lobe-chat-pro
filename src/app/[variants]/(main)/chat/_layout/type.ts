import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  session: ReactNode;
}
