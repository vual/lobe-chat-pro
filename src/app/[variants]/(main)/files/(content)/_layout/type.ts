import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  menu: ReactNode;
  modal: ReactNode;
}
