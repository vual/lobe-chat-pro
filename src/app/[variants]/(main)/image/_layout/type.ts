import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  menu: ReactNode;
  topic: ReactNode;
}
