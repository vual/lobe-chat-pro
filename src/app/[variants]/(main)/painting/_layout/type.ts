import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  palette: ReactNode;
}
