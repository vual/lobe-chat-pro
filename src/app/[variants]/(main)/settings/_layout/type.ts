import { ReactNode } from 'react';

export interface LayoutProps {
  category: ReactNode;
  children: ReactNode;
}
