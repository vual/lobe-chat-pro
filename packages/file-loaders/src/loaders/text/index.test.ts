import path from 'node:path';
import { beforeEach, describe, expect, it } from 'vitest';

import type { FileLoaderInterface } from '../../types';
import { TextLoader } from './index';

const fixturePath = (filename: string) => path.join(__dirname, `./fixtures/${filename}`);

let loader: FileLoaderInterface;

const testFile = fixturePath('test.txt');

beforeEach(() => {
  loader = new TextLoader();
});

describe('TextLoader', () => {
  it('should load pages correctly', async () => {
    const pages = await loader.loadPages(testFile);
    expect(pages).toHaveLength(1);
    const page = pages[0];
    expect(page).toMatchSnapshot();
  });

  it('should aggregate content correctly', async () => {
    const pages = await loader.loadPages(testFile);
    const content = await loader.aggregateContent(pages);
    // Default aggregation joins with newline
    expect(content).toBe('Hello Text.\nSecond Line.\n');
  });

  it('should handle file read errors in loadPages', async () => {
    const pages = await loader.loadPages(fixturePath('nonexistent.txt'));
    expect(pages).toHaveLength(1);
    expect(pages[0].metadata.error).toContain('Failed to load text file');
    expect(pages[0].pageContent).toBe('');
  });
});
