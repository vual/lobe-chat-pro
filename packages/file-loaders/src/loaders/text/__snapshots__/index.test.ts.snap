// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextLoader > should load pages correctly 1`] = `
{
  "charCount": 25,
  "lineCount": 3,
  "metadata": {
    "lineNumberEnd": 3,
    "lineNumberStart": 1,
  },
  "pageContent": "Hello Text.
Second Line.
",
}
`;
